server/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoTileBoundedPredicateTests.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.search.aggregations.bucket.geogrid;
+
+import org.elasticsearch.common.geo.GeoBoundingBox;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.geometry.Rectangle;
+import org.elasticsearch.test.ESTestCase;
+
+public class GeoTileBoundedPredicateTests extends ESTestCase {
+
+    public void testValidTile() {
+        int precision = randomIntBetween(0, GeoTileUtils.MAX_ZOOM);
+        int tiles = 1 << precision;
+        int x = randomIntBetween(0, tiles - 1);
+        int y = randomIntBetween(0, tiles - 1);
+        Rectangle rectangle = GeoTileUtils.toBoundingBox(x, y, precision);
+        GeoBoundingBox bbox = new GeoBoundingBox(
+            new GeoPoint(rectangle.getMaxLat(), rectangle.getMinLon()),
+            new GeoPoint(rectangle.getMinLat(), rectangle.getMaxLon())
+        );
+        long hash = GeoTileUtils.longEncodeTiles(precision, x, y);
+        GeoTileBoundedPredicate predicate = new GeoTileBoundedPredicate(precision, bbox);
+
+        int minX = GeoTileUtils.getXTile(bbox.left(), tiles);
+        int maxX = GeoTileUtils.getXTile(bbox.right(), tiles);
+        int minY = GeoTileUtils.getYTile(bbox.bottom(), tiles);
+        int maxY = GeoTileUtils.getYTile(bbox.top(), tiles);
+
+        assertPredicates(hash, predicate, minX, minY, precision);
```

Comment:
I ran this test and found all four corners tested here result in the predicate returning `false`? Somehow I was expecting at least one of the corners to return true based on the min/max comment in the predicate.

Reply:
In the other tests `GeoHashBoundedPredicateTests.testValidTile` the min/min test (left, bottom) returned true for the predicate. I expected the same here, but it returns false. This seems wrong, surely?

Reply:
GeoTile transformation are not stable, in particular on the latitude because of the trigonometry that we need to use to compute the Y value (maths are base on https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#Java). So it might happen that `tileToLat` gives you a lat, and `getYTile` using that lat gives you a different tile. It is not important here at the moment but we should keep this in mind.